tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -1002,6 +1025,184 @@ describe(__filename, () => {
     expect(root).toHaveState('pictureData', null);
   });
 
+  it('displays a modal when user clicks the delete profile button', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const preventDefaultSpy = sinon.spy();
+
+    const root = renderUserProfileEdit({ store });
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(0);
+
+    root.find('.UserProfileEdit-delete-button').simulate(
+      'click',
+      createFakeEvent({ preventDefault: preventDefaultSpy })
+    );
+
+    sinon.assert.called(preventDefaultSpy);
+
+    const modal = root.find('.UserProfileEdit-deletion-modal');
+
+    expect(modal).toHaveLength(1);
+    expect(modal).toHaveProp(
+      'header',
+      'Attention: you are about to delete your profile. Are you sure?'
+    );
+    expect(modal).toHaveProp('visibleOnLoad', true);
+
+    expect(modal.find('p').at(0)).toHaveProp('dangerouslySetInnerHTML', {
+      __html: oneLine`If you confirm this <strong>irreversible action</strong>,
+        the following data will be removed: profile picture, profile details
+        (including username, email, display name, location, home page,
+        biography, occupation) and notification preferences. Other data such as
+        ratings and reviews will be anonymized.`,
+    });
+
+    expect(modal.find('p').at(1)).toHaveText(oneLine`Important: if you own
+      add-ons, you have to transfer them to other users or to delete them
+      before you can delete your profile.`);
+
+    expect(root.find('.UserProfileEdit-confirm-button')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-confirm-button').children())
+      .toHaveText('Yes, delete my profile');
+    expect(root.find('.UserProfileEdit-cancel-button')).toHaveLength(1);
+  });
+
+  it('renders different information in the modal when user to be deleted is not the current logged-in user', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const params = { username: 'another-user' };
+
+    const root = renderUserProfileEdit({ params, store });
+
+    root.find('.UserProfileEdit-delete-button')
+      .simulate('click', createFakeEvent());
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveProp(
+      'header',
+      'Attention: you are about to delete a profile. Are you sure?'
+    );
+
+    expect(root.find('.UserProfileEdit-deletion-modal').find('p'))
+      .toHaveLength(1);
+
+    expect(root.find('.UserProfileEdit-confirm-button').children())
+      .toHaveText('Yes, delete this profile');
+  });
+
+  it('closes the modal when user clicks the cancel button', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const root = renderUserProfileEdit({ store });
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(0);
+
+    root.find('.UserProfileEdit-delete-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(1);
```

Comment:
It seems like it would be worth including the assertion for `preventDefault` as you did for the delete button on the main form.